CONTRIBUTING.md

```diff
@@ -84,11 +84,11 @@ Mbed TLS is well documented, but if you think documentation is needed, speak out
 License and Copyright
 ---------------------
 
-Unless specifically indicated otherwise in a file, Mbed TLS files are provided under the [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) license. See the [LICENSE](LICENSE) file for the full text of this license.
+Unless specifically indicated otherwise in a file, Mbed TLS files are provided under a dual [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) OR [GPL-2.0-or-later](https://spdx.org/licenses/GPL-2.0-or-later.html) license. See the [LICENSE](LICENSE) file for the full text of these licenses. This means that users may choose which of these licenses they take the code under.
```

Comment:
We are now releasing everything under a dual license except Everest and p256-m. I assume this is deliberate.

Reply:
For p256-m I wanted to check with @mpg first. For Everest, I don't know the situation. Are we permitted to release under GPL?

Reply:
Everest was contributed through https://github.com/Mbed-TLS/mbedtls/pull/2073 (eventually merged via https://github.com/ARMmbed/mbed-crypto/pull/140). The Everest files have always had Apache-only header files. However, to the best of my knowledge, the contributor had signed the Contributor License Agreement (CLA) which allowed Arm to distribute the contribution under any other license at Arm's choice. The fact that there was no discussion of the license during the review is not particularly an indication that there was an agreement that Everest would be Apache-only: given our practice at the time, we may have just assumed that the overall license would apply. If there was any discussion between Arm and Microsoft about the license, I either wasn't in on it or have completely forgotten. We would need to dig through legal documents to figure it out for sure though.

In any case, it's not particularly surprising to recipients that the content of a `3rdparty` repository might have a different license from the rest of the project, and everything under `3rdparty` is optional, so this isn't critical.

Reply:
For p256-m, I'm more than happy to relicense to Apache-or-GPL.

Reply:
Thanks @mpg, I'll update p256-m.

For Everest: we have never distributed it as GPL, so there's no precedent, and it is marked as Apache 2.0. It is definitely quite possible that the authors did not realise/intend for it to be under a GPL license. On that basis I don't feel comfortable putting it under the dual license without consulting the authors. I will leave it as Apache 2.0 unless the situation changes. @wintersteiger - we would be grateful for your input here.